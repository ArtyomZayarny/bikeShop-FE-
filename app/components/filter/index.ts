export * from "./quick-filter-btn/quick-filter-btn";
export * from "./filter";
