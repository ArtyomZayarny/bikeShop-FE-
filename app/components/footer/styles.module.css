.footer {
    background-color: var(--dark);
}
.copyright {
    margin-top:auto
}
.callback {
    border-radius: 4px;
background: linear-gradient(0deg, rgba(255, 255, 255, 0.05) 0%, rgba(255, 255, 255, 0.05) 100%), #2E424D;
box-shadow: 0px 1px 4px 0px rgba(0, 0, 0, 0.20);
}