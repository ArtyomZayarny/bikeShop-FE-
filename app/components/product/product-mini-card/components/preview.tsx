import Image from "next/image";

type Props = {
  imageUrl: string;
};

export const Preview = ({ imageUrl }: Props) => {
  return (
    <div className="flex justify-center">
      <Image src={imageUrl} alt="trek" width={200} height={107} />
    </div>
  );
};
