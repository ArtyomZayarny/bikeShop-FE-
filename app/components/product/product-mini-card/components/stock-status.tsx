type Props = {
  inStock: boolean;
};
export const StockStatus = ({ inStock }: Props) => {
  return (
    <div>
      <span
        className={`text-sm font-normal text-${inStock ? "green" : "disabled"}`}
      >
        {inStock ? "В наявності" : "Немає в наявності"}
      </span>
    </div>
  );
};
