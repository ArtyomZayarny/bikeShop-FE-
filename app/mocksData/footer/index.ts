export const footerLinks = {
  customer: {
    copyRight: "2017-2023 Інтернет магазин велосипедів Jolly Ride",
    title: "Покупцеві",
    items: [
      {
        label: "Про нас",
        href: "./about-us",
      },
      { label: "Оплата і доставка", href: "./dostavka" },
      { label: "Обмін та повернення", href: "./obmen" },
      { label: "Публічний договір оферти прокату", href: "./oferta" },
      { label: "Прокат", href: "./prokat" },
      { label: "Ремонт", href: "./remont" },
      { label: "Блог", href: "./blog" },
    ],
  },
  catalog: {
    title: "Каталог",
    items: [
      { label: "Новинки", href: "./new-prosucts" },
      { label: "Акції", href: "./akcii" },
      { label: "Розпродаж", href: "./sale" },
      { label: "Велосипеди", href: "./bikes" },
      { label: "Аксесуари", href: "./accessories" },
      { label: "Запчастини", href: "./details" },
      { label: "Інструменти та мастила", href: "./instruments" },
      { label: "Велоодяг", href: "./ware" },
      { label: "Термобілизна", href: "./term-ware" },
      { label: "Зимовий спорт", href: "./winter-sport" },
    ],
  },
  contacts: {
    items: [
      { label: "address", value: "м. Івано-Франківськ, вул. Галицька 145 Б" },
      { label: "workTime", value: "Пн-Сб: 10:00-19:00" },
      {
        label: "email",
        value: "jollyride.if@gmail.com",
        icon: "./svg/email.svg",
      },
      {
        label: "phone",
        value: "+380962071551",
        formattedValue: "+38 (096) 207 15 51",
        icon: "./svg/phone.svg",
      },
    ],
    socials: [
      {
        label: "instagram",
        href: "https://www.instagram.com/jolly.ride",
      },
      {
        label: "facebook",
        href: "https://www.facebook.com/JollyRideShop",
      },
      {
        label: "telegram",
        href: "https://t.me/veloif",
      },
      {
        label: "viber",
        href: "https://www.viber.com/",
      },
      {
        label: "tik_tok",
        href: "https://www.tiktok.com/@jollyride1?_t=8bEmdEx0CH8&_r=1",
      },
    ],
  },
};
