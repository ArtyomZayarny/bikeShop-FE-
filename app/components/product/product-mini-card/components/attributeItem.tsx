import { DropDown } from "../../../common/dropdown";
import { Title } from "../../../common/title";
import { ColorList } from "../../../filter/color-list";
import { filterBlock } from "../../../types";

type Props = {
  attr: filterBlock;
};
export const AttributeItem = ({ attr }: Props) => {
  const renderAttributeItem = () => {
    switch (attr.type) {
      case "color":
        return (
          <>
            <ColorList colors={attr.items!} />
          </>
        );
      case "size":
        return (
          <>
            <DropDown options={attr.items!} />
          </>
        );
    }
  };
  return (
    <li className="flex mb-4 flex-col">
      <Title type={6} styles="text-sm font-normal mb-2">
        {attr.label}
      </Title>
      {renderAttributeItem()}
    </li>
  );
};
