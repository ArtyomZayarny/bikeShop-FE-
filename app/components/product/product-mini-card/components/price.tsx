import { PriceType } from "@/app/components/types";

type Props = {
  price: PriceType;
};

export const Price = ({ price }: Props) => {
  return (
    <div>
      {/* price */}
      <span className="text-xs text-regular font-semibold">
        {price.newPrice} грн
      </span>
      {/* old price */}
      <span className="text-xs ml-2 text-disabled line-through">
        {price.oldPrice} грн
      </span>
    </div>
  );
};
