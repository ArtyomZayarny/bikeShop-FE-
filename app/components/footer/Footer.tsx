import Link from "next/link";
import { Container } from "../container/container";
import Image from "next/image";
import styles from "./styles.module.css";
import { footerLinks } from "../../mocksData/footer";
import { Title } from "../common/title";

export const Footer = () => {
  return (
    <footer className={`flex justify-center ${styles.footer} py-10`}>
      <Container styles="items-stretch">
        {/* For Customer*/}
        <div className="flex flex-col">
          <Title type={3} styles="text-white text-base font-bold mb-4">
            {footerLinks.customer.title}
          </Title>
          <ul className="flex flex-col h-full">
            {footerLinks.customer.items.map((item) => (
              <li key={item.label}>
                <Link href={item.href} className="text-white text-sm">
                  {item.label}
                </Link>
              </li>
            ))}
            <li className={`text-white text-sm ${styles.copyright}`}>
              ©{footerLinks.customer.copyRight}
            </li>
          </ul>
        </div>

        {/* Catalog */}
        <div>
          <Title type={3} styles="text-white text-base font-bold mb-4">
            {footerLinks.catalog.title}
          </Title>
          <ul>
            {footerLinks.catalog.items.map((item) => (
              <li key={item.label}>
                <Link href={item.href} className="text-white text-sm">
                  {item.label}
                </Link>
              </li>
            ))}
          </ul>
        </div>

        {/* Contacts */}
        <div className="flex flex-col items-start">
          <ul>
            {footerLinks.contacts.items.map((item) => (
              <li
                key={item.label}
                className="text-white text-sm pb-4  last:p:0"
              >
                {item?.icon ? (
                  <div className="flex items-center">
                    <Image
                      className="mr-2"
                      src={item.icon}
                      alt={item.label}
                      width={24}
                      height={24}
                    />
                    <a
                      className="flex"
                      href={
                        item.label === "phone"
                          ? `tel:${item.value}`
                          : `mailto:${item.value}`
                      }
                    >
                      {item?.formattedValue || item.value}
                    </a>
                  </div>
                ) : (
                  item.value
                )}
              </li>
            ))}
          </ul>
          <button className={`${styles.callback} py-2 px-3`}>
            <span className="text-white text-xs font-medium">
              ПЕРЕДЗВОНИТИ МЕНІ
            </span>
          </button>
          <ul className="flex gap-4 mt-auto">
            {footerLinks.contacts.socials.map((item) => (
              <li key={item.label}>
                <Link href={item.href}>
                  <Image
                    src={`./svg/${item.label}.svg`}
                    alt={item.label}
                    width={24}
                    height={24}
                  />
                </Link>
              </li>
            ))}
          </ul>
        </div>
      </Container>
    </footer>
  );
};
