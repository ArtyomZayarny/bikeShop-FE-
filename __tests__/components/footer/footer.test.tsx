import { render } from "@testing-library/react";
import { Footer } from "../../../app/components/footer";

describe("Footer", () => {
  it("should render footer", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
