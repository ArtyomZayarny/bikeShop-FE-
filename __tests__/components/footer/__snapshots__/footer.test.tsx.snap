// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render footer 1`] = `
<div>
  <footer
    class="flex justify-center footer py-10"
  >
    <div
      class="justify-between py-6 w-full 2xl:max-w-screen-2xl xl:max-w-screen-xl lg:px-10 lg:flex md:px-5 items-stretch"
    >
      <div
        class="flex flex-col"
      >
        <h3
          class="text-white text-base font-bold mb-4"
        >
          Покупцеві
        </h3>
        <ul
          class="flex flex-col h-full"
        >
          <li>
            <a
              class="text-white text-sm"
              href="./about-us"
            >
              Про нас
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./dostavka"
            >
              Оплата і доставка
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./obmen"
            >
              Обмін та повернення
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./oferta"
            >
              Публічний договір оферти прокату!
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./prokat"
            >
              Прокат
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./remont"
            >
              Ремонт
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./blog"
            >
              Блог
            </a>
          </li>
          <li
            class="text-white text-sm copyright"
          >
            ©
            2017-2023 Інтернет магазин велосипедів Jolly Ride
          </li>
        </ul>
      </div>
      <div>
        <h3
          class="text-white text-base font-bold mb-4"
        >
          Каталог
        </h3>
        <ul>
          <li>
            <a
              class="text-white text-sm"
              href="./new-prosucts"
            >
              Новинки
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./akcii"
            >
              Акції
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./sale"
            >
              Розпродаж
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./bikes"
            >
              Велосипеди
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./accessories"
            >
              Аксесуари
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./details"
            >
              Запчастини
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./instruments"
            >
              Інструменти та мастила
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./ware"
            >
              Велоодяг
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./term-ware"
            >
              Термобілизна
            </a>
          </li>
          <li>
            <a
              class="text-white text-sm"
              href="./winter-sport"
            >
              Зимовий спорт
            </a>
          </li>
        </ul>
      </div>
      <div
        class="flex flex-col items-start"
      >
        <ul>
          <li
            class="text-white text-sm pb-4  last:p:0"
          >
            м. Івано-Франківськ, вул. Галицька 145 Б
          </li>
          <li
            class="text-white text-sm pb-4  last:p:0"
          >
            Пн-Сб: 10:00-19:00
          </li>
          <li
            class="text-white text-sm pb-4  last:p:0"
          >
            <div
              class="flex items-center"
            >
              <img
                alt="email"
                class="mr-2"
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="./svg/email.svg"
                style="color: transparent;"
                width="24"
              />
              <a
                class="flex"
                href="mailto:jollyride.if@gmail.com"
              >
                jollyride.if@gmail.com
              </a>
            </div>
          </li>
          <li
            class="text-white text-sm pb-4  last:p:0"
          >
            <div
              class="flex items-center"
            >
              <img
                alt="phone"
                class="mr-2"
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="./svg/phone.svg"
                style="color: transparent;"
                width="24"
              />
              <a
                class="flex"
                href="tel:+380962071551"
              >
                +38 (096) 207 15 51
              </a>
            </div>
          </li>
        </ul>
        <button
          class="callback py-2 px-3"
        >
          <span
            class="text-white text-xs font-medium"
          >
            ПЕРЕДЗВОНИТИ МЕНІ
          </span>
        </button>
        <ul
          class="flex gap-4 mt-auto"
        >
          <li>
            <a
              href="https://www.instagram.com/jolly.ride"
            >
              <img
                alt="instagram"
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="./svg/instagram.svg"
                style="color: transparent;"
                width="24"
              />
            </a>
          </li>
          <li>
            <a
              href="https://www.facebook.com/JollyRideShop"
            >
              <img
                alt="facebook"
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="./svg/facebook.svg"
                style="color: transparent;"
                width="24"
              />
            </a>
          </li>
          <li>
            <a
              href="https://t.me/veloif"
            >
              <img
                alt="telegram"
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="./svg/telegram.svg"
                style="color: transparent;"
                width="24"
              />
            </a>
          </li>
          <li>
            <a
              href="https://www.viber.com/"
            >
              <img
                alt="viber"
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="./svg/viber.svg"
                style="color: transparent;"
                width="24"
              />
            </a>
          </li>
          <li>
            <a
              href="https://www.tiktok.com/@jollyride1?_t=8bEmdEx0CH8&_r=1"
            >
              <img
                alt="tik_tok"
                data-nimg="1"
                decoding="async"
                height="24"
                loading="lazy"
                src="./svg/tik_tok.svg"
                style="color: transparent;"
                width="24"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </footer>
</div>
`;
